File: cpp/1813. 句子相似性 III.cpp
class Solution {
public:
    bool areSentencesSimilar(string sentence1, string sentence2) {
        vector<string> words1,words2;
        string t="";
        for(char ch:sentence1){
            if(ch!=' '){
                t+=ch;
            }else{
                words1.emplace_back(t);
                t="";
            }
        }
        words1.emplace_back(t);
        t="";
        for(char ch:sentence2){
            if(ch!=' '){
                t+=ch;
            }else{
                cout<<t<<endl;
                words2.emplace_back(t);
                t="";
            }
        }
        cout<<t<<endl;
        words2.emplace_back(t);
        t="";
        int n1=words1.size(),n2=words2.size();
        int i=0,j=0;
        while(i<n1&&i<n2&&words1[i]==words2[i]){
            i++;
        }
        while(j<n1-i&&j<n2-i&&words1[n1-j-1]==words2[n2-j-1]){
            j++;
        }
        return i+j==min(n1,n2);
    }
};

File: 刷题/golang/1813. 句子相似性 III.go
func areSentencesSimilar(sentence1 string, sentence2 string) bool {
    words1:=strings.Split(sentence1," ")
    words2:=strings.Split(sentence2," ")
    i,n:=0,len(words1)
    j,m:=0,len(words2)
    for i<n&&i<m&&words1[i]==words2[i]{
        i++
    }
    for j<n-i&&j<m-i&&words1[n-j-1]==words2[m-j-1]{
        j++
    }
    return i+j==min(n,m)
}
func min(x,y int) int{
    if x<y {
        return x
    }
    return y
}